### DISCUSSION TITLE
What is the difference of batch size and the number of seed nodes?

### DISCUSSION THREAD

**Discussion by Gibyeng** [pyg-team/pyg-lib#301] [General]
What is the relationship of batch size and the number of seed nodes in NeighborSampler?
To my understanding, the batch size is equal to the number of seed nodes, which means the first nodes in the batch.
However I found the two variables "seed.size(0)" (in .neighbor_kernel.cpp) and "batch_size" (in .py file) have different values.

**Comment by rusty1s** [comment]
`seed` refers to the input nodes you want to process in one epoch, and `batch_size` is determines how to divide these input nodes into different chunks.

**Reply by Gibyeng** [reply]
@rusty1s Could you explain it with an example, say if seed.size(0) is 207 and batch_size is 512. Does it mean we need two epoches to collect 512 samples in total to form a batch?

**Reply by rusty1s** [reply]
The last batch doesn't need to have full capacity. If you have `seed.size() == 207` and `batch_size = 512`, then all 207 seed nodes will be handled within a single batch.

**Reply by Gibyeng** [reply]
@rusty1s  Thanks a lot, I got it :)
